code/src/components/app-button/AppButton.types.ts

```diff
@@ -10,8 +11,15 @@ type ButtonVariant =
   | "shadow";
 type ButtonSize = "small" | "medium" | "large" | "extra-large";
 
-export type AppButtonProps = Omit<ButtonProps, "variant" | "size"> & {
+type BaseProps = Omit<ButtonProps, "variant" | "size"> & {
   variant?: ButtonVariant;
   size?: ButtonSize;
   isLoading?: boolean;
 };
+
+export type AppButtonProps = BaseProps &
+  (
+    | ({ isNavLink: true } & NavLinkProps)
+    | ({ isNavLink: false } & LinkProps)
+    | { isNavLink?: boolean; to?: To }
+  );
```

Comment:
Try to reuse `AppLinkProps`:

```jsx
export type AppLinkProps = Omit<MuiLinkProps, "variant" | "component"> &
  BaseProps &
  (
    | ({ isNavLink: true } & NavLinkProps)
    | ({ isNavLink?: false } & ReactRouterDomLinkProps)
  );
```

Reply:
Thanks, I created reusable type
```
type AppLinkWithNavLinkProps =
  | ({ isNavLink: true } & NavLinkProps)
  | ({ isNavLink?: false } & ReactRouterDomLinkProps)
```
I think that is enough because slightly different logic (for example, to must be present in `AppLink` but optional in `AppButton`) requires complex types to be created to make it more reusable.